sematic/ui/packages/common/src/component/DateTime.tsx

```diff
@@ -31,9 +37,15 @@ export const Duration = (start: Date | string, end: Date | string) => {
     const duration = intervalToDuration({
         start: startTime,
         end: endTime
-    })
+    });
+
+    const formatString = formatDuration(duration, { format: ["minutes", "seconds"], zero: true});
+
+    return formatString.replace(/\s0\sseconds$/g, "");
+}
 
-    return formatDuration(duration, { format: ["minutes", "seconds"]});
+export function DurationShort(start: Date | string, end: Date | string) {
+    return Duration(start, end).replace(/minutes/g, "m").replace(/seconds/g, "s");
```

Comment:
Shouldn't these parameters always be `Date`s instead of `string`s, in order to avoid this ugly post-processing? Where do the strings come from?

Reply:
a string is the data string from the JSON payload. In a JSON payload, there is no easy way to represent typed Date, only strings are commonly adopted. ([Reference](https://stackoverflow.com/questions/10286204/what-is-the-right-json-date-format)). Separately, I used the `Date` | 'string' type hoping the utility function can be more generally applicable.

Also, I am not sure if the comment is placed at the right spot -- this function (`.replace(/minutes/g, "m").replace(/seconds/g, "s")`) is to convert formatted duration from long form "x minutes y seconds" to the short form " x m y s") to save space in UI. It doesn't deal with Date vs string types for timestamps. Is this part ugly? If so, what do you suggest to improve?

Reply:
The type is represented as `Date | string`. `Date` is programmatic, and can be used to generate the human-readable string in the way the developer intended.
Using the `string` option requires the hacky substitution.

If the field comes from a Server call, without it being a `Date`, shouldn't the type of the parameter be just `string` instead?

"The right way to do it" would be to convert the `string` to a `Data`, then generate the date string to use here prgrammaticaly, but that would be overkill, so we should avoid it.

Reply:
Thanks for the discussion. I ended up using `Date` as the parameter type which allows converters to deal with parameter conversions.